src/backend/dx12/src/command.rs

```diff
@@ -602,6 +616,148 @@ impl CommandBuffer {
+                    z: gap / slice_pitch as i32,
+                };
+                let footprint = image::Extent {
+                    width: buf_offset.x as u32 + r.image_extent.width,
+                    height: buf_offset.y as u32 + r.image_extent.height,
+                    depth: buf_offset.z as u32 + r.image_extent.depth,
+                };
+                if r.image_extent.width + buf_offset.x as u32 <= row_pitch_texels {
+                    // we can map it to the aligned one and adjust the offsets accordingly
+                    copies.push(Copy {
+                        footprint_offset: aligned_offset,
+                        footprint,
+                        row_pitch,
+                        img_subresource,
+                        img_offset: r.image_offset,
+                        buf_offset,
+                        copy_extent: r.image_extent,
+                    });
+                } else {
+                    // split the copy region into 2 that suffice the previous condition
+                    assert!(buf_offset.x as u32 <= row_pitch_texels);
+                    let half = row_pitch_texels - buf_offset.x as u32;
+                    assert!(half <= r.image_extent.width);
+
+                    copies.push(Copy {
+                        footprint_offset: aligned_offset,
+                        footprint: image::Extent {
+                            width: row_pitch_texels,
+                            .. footprint
+                        },
+                        row_pitch,
+                        img_subresource,
+                        img_offset: r.image_offset,
+                        buf_offset,
+                        copy_extent: image::Extent {
+                            width: half,
+                            .. r.image_extent
+                        },
+                    });
+                    copies.push(Copy {
+                        footprint_offset: aligned_offset,
+                        footprint: image::Extent {
+                            width: r.image_extent.width - half,
+                            height: footprint.height + 1,
+                            depth: footprint.depth,
+                        },
+                        row_pitch,
+                        img_subresource,
+                        img_offset: image::Offset {
+                            x: r.image_offset.x + half as i32,
+                            .. r.image_offset
+                        },
+                        buf_offset: image::Offset {
+                            x: 0,
+                            .. buf_offset
+                        },
+                        copy_extent: image::Extent {
+                            width: r.image_extent.width - half,
+                            .. r.image_extent
+                        },
+                    });
+                }
+            } else {
+                // worst case: row by row copy
```

Comment:
Could you explain the idea here?
Imo the worst case scenario here is a image laid out in memory like this, considering row pitch alignment and data alignment:
```
-----------
----------x
xxxxxxxxxoo
ooooooooyyy
yyyyyyyzzzz
zzzz ...
```
The resulting image should be
```
xxxxxxxxxx
oooooooooo
yyyyyyyyyy
zzzzzzzzzz
...
```